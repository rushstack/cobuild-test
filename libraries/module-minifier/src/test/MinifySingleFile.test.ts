// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import { minifySingleFileAsync } from '../MinifySingleFile';

describe(minifySingleFileAsync.name, () => {
  it('uses consistent identifiers for webpack vars', async () => {
    const code: string = `__MINIFY_MODULE__(function (module, __webpack_exports__, __webpack_require__) {});`;

    const minifierResult = await minifySingleFileAsync(
      {
        hash: 'foo',
        code,
        nameForMap: undefined,
        externals: undefined
      },
      {
        mangle: true
      }
    );

    expect(minifierResult).toMatchSnapshot();
  });
});
