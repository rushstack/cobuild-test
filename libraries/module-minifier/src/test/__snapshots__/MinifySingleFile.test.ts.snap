// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`minifySingleFileAsync uses consistent identifiers for webpack vars 1`] = `
Object {
  "code": "__MINIFY_MODULE__((function(e,t,n){}));",
  "error": undefined,
  "hash": "foo",
  "map": undefined,
}
`;
